backend/danswer/connectors/confluence/connector.py

```diff
@@ -286,8 +314,73 @@ def _fetch(start_ind: int, batch_size: int) -> Collection[dict[str, Any]]:
 
                 return view_pages
 
+        def _fetch_child_pages(
```

Comment:
this seems like it only goes one level deep. If I imagine what someone wants to do when specifying a page, I feel like they'd generally want to index a whole "subspace" under a certain page. 

Maybe this could be pulled out into a separate `ConfluenceRecursivePageFetcher` which maintains its own state of possible child pages, etc. 🤔